packages/rule-stylesheet-limits/src/rule.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * @fileoverview Checks if CSS exceeds known stylesheet limits
+ */
+
+import { Category } from 'sonarwhal/dist/src/lib/enums/category';
+import { RuleContext } from 'sonarwhal/dist/src/lib/rule-context';
+// The list of types depends on the events you want to capture.
+import { IRule, ScanEnd, RuleMetadata } from 'sonarwhal/dist/src/lib/types';
+import { debug as d } from 'sonarwhal/dist/src/lib/utils/debug';
+import { RuleScope } from 'sonarwhal/dist/src/lib/enums/rulescope';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class StylesheetLimitsRule implements IRule {
+
+    public static readonly meta: RuleMetadata = {
+        docs: {
+            category: Category.interoperability,
+            description: `Checks if CSS exceeds known stylesheet limits`
+        },
+        id: 'stylesheet-limits',
+        schema: [{
+            additionalProperties: false,
+            properties: {
+                maxRules: { type: 'number' },
+                maxSheets: { type: 'number' },
+                maxImports: { type: 'number' }
+            },
+            type: ['object', 'null']
+        }],
+        scope: RuleScope.any
+    }
+
+    public constructor(context: RuleContext) {
+
+        const includesIE9 = context.targetedBrowsers.includes('ie 9');
+
+        let maxRules = includesIE9 ? 4095 : 65534;
+        let maxSheets = includesIE9 ? 31 : 4095;
+        let maxImports = includesIE9 ? 4 : 0;
+
+        //  allow limits to be overridden by rule options
+        if (context.ruleOptions) {
```

Comment:
I'm not sure about this approach. I can see someone targetting IE9 increasing the number of `maxRules` to 5000 which is not great instead of removing the browser from their `browserslist`.
Maybe we should do a `Math.min()`?